src/api/stats.ts

```diff
@@ -130,70 +132,89 @@ export const convertToUTC = (
 //  Right now all tables run the same query even though they only need
 //  2 - 4 columns. However, not a huge deal perf wise because the other cols
 //  are all 0.
-const getStatsByName = (names: string[], filter?: StatsFilter) => {
-    let queryBuilder = supabaseClient
-        .from<CatalogStats>(TABLES.CATALOG_STATS)
-        .select(DEFAULT_QUERY)
-        .in('catalog_name', names)
-        .order('catalog_name');
-
-    const today = new Date();
-
-    // TODO (locale) allow users to have proper locale settings used for start and end of weeks
-    // startOf/endOf functions can give some odd results so just forcing exactly
-    //  what days we want to say are the start and end of a week based on the
-    //  current day.
-    const weekStart = isSunday(today) ? today : previousSunday(today);
-    const weekEnd = isSaturday(today) ? today : nextSaturday(today);
-
-    switch (filter) {
-        // Day Range
-        case 'today':
-            queryBuilder = queryBuilder
-                .eq('ts', convertToUTC(today, dailyGrain))
-                .eq('grain', dailyGrain);
-            break;
-        case 'yesterday':
-            queryBuilder = queryBuilder
-                .eq('ts', convertToUTC(subDays(today, 1), dailyGrain))
-                .eq('grain', dailyGrain);
-            break;
-
-        // Week Range
-        case 'thisWeek':
-            queryBuilder = queryBuilder
-                .gte('ts', convertToUTC(weekStart, dailyGrain))
-                .lte('ts', convertToUTC(weekEnd, dailyGrain))
-                .eq('grain', dailyGrain);
-            break;
-        case 'lastWeek':
-            queryBuilder = queryBuilder
-                .gte('ts', convertToUTC(subWeeks(weekStart, 1), dailyGrain))
-                .lte('ts', convertToUTC(subWeeks(weekEnd, 1), dailyGrain))
-                .eq('grain', dailyGrain);
-            break;
-
-        // Month Range
-        case 'thisMonth':
-            queryBuilder = queryBuilder
-                .eq('ts', convertToUTC(today, monthlyGrain))
-                .eq('grain', monthlyGrain);
-
-            break;
-        case 'lastMonth':
-            queryBuilder = queryBuilder
-                .eq('ts', convertToUTC(subMonths(today, 1), monthlyGrain))
-                .eq('grain', monthlyGrain);
-            break;
-
-        default:
-            throw new Error('Unsupported filter used in Stats Query');
+const getStatsByName = async (names: string[], filter?: StatsFilter) => {
+    const limiter = pLimit(3);
+    const promises: Array<Promise<PostgrestResponse<CatalogStats>>> = [];
+    let index = 0;
+
+    // TODO (retry) promise generator
+    const promiseGenerator = (idx: number) => {
```

Comment:
The bulk of this change is just moving the previous function inside of the `promiseGenerator`